optimum/neuron/pipelines/transformers/base.py

```diff
@@ -0,0 +1,238 @@
+                    "feature_extractor explictly"
+                )
+        model_id = None
+    else:
+        raise ValueError(
+            f"""Model {model} is not supported. Please provide a valid model either as string or NeuroModel.
+            You can also provide non model then a default one will be used"""
+        )
+    return model, model_id, tokenizer, feature_extractor
+
+
+def pipeline(
+    task: str = None,
+    model: Optional[Any] = None,
+    tokenizer: Optional[Union[str, PreTrainedTokenizer]] = None,
+    feature_extractor: Optional[Union[str, PreTrainedFeatureExtractor]] = None,
+    use_fast: bool = True,
+    export: bool = False,
+    input_shapes: Optional[Dict[str, int]] = None,
+    use_auth_token: Optional[Union[str, bool]] = None,
+    revision: Optional[str] = None,
+    trust_remote_code: Optional[bool] = None,
+    *model_kwargs,
+    **kwargs,
+) -> Pipeline:
+    if task not in list(NEURONX_SUPPORTED_TASKS.keys()):
+        raise ValueError(
+            f"Task {task} is not supported for the optimum neuron pipeline. Supported tasks are {list(NEURONX_SUPPORTED_TASKS.keys())}"
+        )
+
+    if input_shapes is None and export:
+        input_shapes = {"batch_size": 1, "sequence_length": 128}
+        logger.warning(f"No input shapes provided, using default shapes, {input_shapes}")
+
+    # copied from transformers.pipelines.__init__.py
+    hub_kwargs = {
+        "revision": revision,
+        "use_auth_token": use_auth_token,
+        "trust_remote_code": trust_remote_code,
+        "_commit_hash": None,
+    }
+
+    config = kwargs.get("config", None)
+    if config is None and isinstance(model, str):
+        config = AutoConfig.from_pretrained(model, _from_pipeline=task, **hub_kwargs, **kwargs)
+        hub_kwargs["_commit_hash"] = config._commit_hash
+
+    no_feature_extractor_tasks = set()
+    no_tokenizer_tasks = set()
+    for _task, values in NEURONX_SUPPORTED_TASKS.items():
+        if values["type"] == "text":
+            no_feature_extractor_tasks.add(_task)
+        elif values["type"] in {"image", "video"}:
+            no_tokenizer_tasks.add(_task)
+        elif values["type"] in {"audio"}:
+            no_tokenizer_tasks.add(_task)
+        elif values["type"] not in ["multimodal", "audio", "video"]:
+            raise ValueError(f"SUPPORTED_TASK {_task} contains invalid type {values['type']}")
+
+    # copied from transformers.pipelines.__init__.py l.609
+    if task in no_tokenizer_tasks:
+        # These will never require a tokenizer.
+        # the model on the other hand might have a tokenizer, but
+        # the files could be missing from the hub, instead of failing
+        # on such repos, we just force to not load it.
+        load_tokenizer = False
+    else:
+        load_tokenizer = True
+
+    if task in no_feature_extractor_tasks:
+        load_feature_extractor = False
+    else:
+        load_feature_extractor = True
```

Comment:
```suggestion
    load_feature_extractor = task not in no_feature_extractor_tasks
```

Reply:
would keep it more readable.

Reply:
I personally find it more readable but that's a matter of taste, it was just a nit, either way is ok.